src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -54,45 +54,51 @@ public class TDTM_ObjectDataGateway implements TDTM_iTableDataGateway {
             return true;    
         return false;
     }
-    
+
     /*******************************************************************************************************
      * @description Returns the classes that will be called for the specified ojbect and action. Returned list 
-     *              is sorted by load order.
+     *              is sorted by load order. Filters out in-active handlers and any where the list of 
+     *              excluded usernames includes the current context user.
      * @param objectName The name of the object.
      * @param action The action (before insert, after insert,...). 
      * @return List<SObject> List of classes to call, sorted by load order.
      **/
     public static List<SObject> getClassesToCallForObject(String objectName, TDTM_Runnable.Action action) {   
         string strAction = action.Name();
+        string currUserName = UserInfo.getUserName();
         List<SObject> listClasses = new list<Sobject>();
         for (Trigger_Handler__c th : listTH) {
-            if (th.Object__c == objectName && th.Trigger_Action__c.contains(strAction)) 
+            Set<String> excludedUserNames = (th.Exclude_For_These_Usernames__c != null ? 
+                    new Set<String>(th.Exclude_For_These_Usernames__c.split(';')) : new Set<String>());
+            if (th.Object__c == objectName && th.Trigger_Action__c.contains(strAction) && th.Active__c == true
+                    && (th.Exclude_For_These_Usernames__c == null || !excludedUserNames.contains(currUserName))) {
```

Comment:
I think we can leave out the null check and the OR since an empty set won't ever contain the current username.